keps/sig-auth/3221-structured-authorization-configuration/README.md

```diff
@@ -0,0 +1,733 @@
+authorizers using a config file, they need to make sure the config file exists before
+upgrading the cluster. Similarly when downgrading clusters, they would need to add
+the flags back to their bootstrap mechanism.
+
+### Version Skew Strategy
+
+Not applicable.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate
+  - Feature gate name: `AuthorizationConfigFromFile`
+  - Components depending on the feature gate:
+        - kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+No.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, the feature can be disabled once enabled.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+When the feature is re-enabled, `--authorization-config-file` flag should be present.
+The behaviour is the same as when the feature is enabled for the first time.
+
+###### Are there any tests for feature enablement/disablement?
+
+> //TODO: How do we ensure functionality in the absence of e2e-framework not having support for feature gates?
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+A rollout can fail when the authorization configuration file being passed doesn't
+exist or is invalid.
+
+Already running workloads won't be impacted but cluster users won't be able to
+access the control plane if the cluster is single-node.
+
+###### What specific metrics should inform a rollback?
+
+Not applicable.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+TBD.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+None.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+> //TODO: To be elaborated more during Beta graduation.
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The cluster administrators can check the flags passed to the `kube-apiserver` if
+they have access to the control plane nodes. If the `--authorization-config-file`
+is set to a valid authorization configuration file, the feature is being used.
+Or, they can look at the metrics exposed by `kube-apiserver`.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Other
+  - Details: They can look at the metrics if `apiserver_authorization_step_invocations_total` is increasing.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+The amount of errors denoted by `apiserver_authorization_step_webhook_error_total`
```

Comment:
```suggestion
The total number of `apiserver_authorization_step_webhook_error_total` metric with failure code
```

Reply:
Note for myself: Revisiting this after resolving other comments around the metric.